opsdroid/skill.py

```diff
@@ -0,0 +1,37 @@
+"""Base class for class-based skills
+"""
+from opsdroid.matchers import add_skill_attributes
+
+class Skill:
+    def __init__(self, opsdroid, config, *args, **kwargs):
+        super(Skill, self).__init__()
+
+        self.opsdroid = opsdroid
+        self.config = config
+
+        for attr in self.__dir__():
+            try:
+                method = getattr(self, attr)
+            except Exception:
+                continue
+
+            for decor_params in getattr(method, '__matchers__', []):
```

Comment:
Could you explain why this is necessary.

Reply:
There can be extreme cases when `getattr` will throw an exception, like properties that need some initialisation.